modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -505,26 +506,11 @@ function Filters({fdsFilterClientExtensions, fdsView, namespace}: IProps) {
 				})),
 			];
 
-			if (fdsView.fdsFiltersOrder) {
-				const order = fdsView.fdsFiltersOrder.split(',');
-
-				let notOrdered: FilterCollection = [];
-
-				notOrdered = filtersOrdered.filter(
-					(filter) => !order.includes(String(filter.id))
-				);
-
-				filtersOrdered = fdsView.fdsFiltersOrder
-					.split(',')
-					.map((fdsFilterId) =>
-						filtersOrdered.find(
-							(filter) => filter.id === Number(fdsFilterId)
-						)
-					)
-					.filter(Boolean) as FilterCollection;
-
-				filtersOrdered = [...notOrdered, ...filtersOrdered];
-			}
+			filtersOrdered = sortItems(
+				filtersOrdered,
+				fdsView.fdsFiltersOrder,
```

Comment:
Similar adjustment needed here too. `fdsFiltersOrder` needs to come from the `responseJSON` since the `fdsView` is never re-fetched if you were to switch between tabs.

```
responseJSON.fdsFiltersOrder
```

This is the response:

<img width="777" alt="Screenshot 2024-05-01 at 10 47 04 AM" src="https://github.com/liferay-frontend/liferay-portal/assets/4496722/fb71e35d-5fb2-4d0a-aa95-7d0f1ae58792">